Problem6/src/problem6/Problem6.java

```diff
@@ -10,7 +19,140 @@
      * @param args the command line arguments
      */
     public static void main(String[] args) {
-        // TODO code application logic here
+        File file = new File("myMatrices.bin");
+
+        FileInputStream fileStream;
+        try {
+            fileStream = new FileInputStream(file);
+        } catch (FileNotFoundException ex) {
+            Logger.getLogger(Problem6.class.getName()).log(Level.SEVERE, null, ex);
+            return;
+        }
+        ObjectInputStream objectStream;
+        try {
+            objectStream = new ObjectInputStream(fileStream);
+        } catch (IOException ex) {
+            Logger.getLogger(Problem6.class.getName()).log(Level.SEVERE, null, ex);
+            try {
+                fileStream.close();
+            } catch (IOException ex1) {
+                Logger.getLogger(Problem6.class.getName()).log(Level.SEVERE, null, ex1);
+            }
+            return;
+        }
+
+        int[][] matrix1;
+        int[][] matrix2;
+        
+        try {
+            matrix1 = (int[][]) objectStream.readObject();
+            matrix2 = (int[][]) objectStream.readObject();
+        } catch (IOException ex) {
+            Logger.getLogger(Problem6.class.getName()).log(Level.SEVERE, null, ex);
+            
+            try {
+                objectStream.close();
+                fileStream.close();
```

Comment:
Once you have successfully constructed the ObjectInputStream object, you don’t need to worry about closing the FileInputStream object (closing the ObjectInputStream object will handle this for you)